File: src/Tests4/Straight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tests4
{
    class Straight : Curve
    {
        double p1, p2;

        public Straight(double x, double y, double p1, double p2) : base(x, y)
        {
            this.p1 = p1;
            this.p2 = p2;
        }

        public double getP1()
        {
            return p1;
        }

        public double getP2()
        {
            return p2;
        }

        public override bool ClosedCurve()
        {
            return false;
        }

        public override Point PointCalcul(int t)
        {
            double x0 = Point.getX() - p1 * t;
            double y0 = Point.getY() - p2 * t;
            return new Point(x0, y0);
        }

        public override double TangentCalcul(int t)
        {
            Console.WriteLine("сама себе касательная");
            return 0;
        }
    }
}

File: src/Straight.cpp
#include <iostream>

#include "Curve.h"
#include "Vector.h"
#include "Point.h"
#pragma once

using namespace std;

class Straight : public Curve
{
private:
	double dot1, dot2;

public:
	Straight(double x, double y, double _dot1, double _dot2)
	: Curve(x,y)
	{
		dot1 = _dot1;
		dot2= _dot2;
	};
	bool ClosedCurve() override
	{
		return false;
	};
	double getDot1()
	{
		return dot1;
	};
	double getDot2()
	{
		return dot2;
	};
	Point* PointCalcul(double t)
	{
		double x = point->getX() - dot1*t;
		double y = point->getY() - dot2*t;
		return new Point(x, y);
	};

	/*Point* NormalVector()
	{

	}*/

	Point* TangentCalcul(double t)
	{
		cout << "касательная к прямой - " << endl;
		return new Point(0, 0);
	};
};